packages/flutter/lib/src/material/data_table.dart

```diff
@@ -879,31 +893,36 @@ class DataTable extends StatelessWidget {
 
     String semanticsLabel = '';
     if (rowHeading != null) {
-      semanticsLabel += '$rowHeading,';
+      semanticsLabel += '$rowHeading, ';
     }
     if (columnHeading != null) {
-      semanticsLabel += '$columnHeading,';
+      semanticsLabel += '$columnHeading, ';
     }
-    label = Container(
-      padding: padding,
-      height: effectiveDataRowHeight,
-      alignment: numeric ? Alignment.centerRight : AlignmentDirectional.centerStart,
-      child: Stack(
-        children: <Widget>[
-          Semantics(
-            label: semanticsLabel,
-            onTapHint: 'Select Row',
-            child: DefaultTextStyle(
-              style: effectiveDataTextStyle.copyWith(
-                color: placeholder ? effectiveDataTextStyle.color!.withOpacity(0.6) : null,
+    final MaterialLocalizations localizations = MaterialLocalizations.of(context);
+    final String tableCoordinatesLabel = localizations.tableCoordinatesLabel('$rowNumber', '$totalRows', '$columnNumber', '$totalColumns');
```

Comment:
given there are 3 `Semantics` here, it would be useful to add comments explaining what each one does/ why is necessary

Reply:
Thanks! Added comments to explain what each one of them does.